python/cuspatial/cuspatial/geometry/geoseries.py

```diff
@@ -0,0 +1,792 @@
+        """
+        A cudf.Index object. A mapping of row-labels.
+        """
+        return self._index
+
+    @index.setter
+    def index(self, index):
+        self._index = index
+
+    def __getitem__(self, item):
+        """
+        Returns gpuGeometry objects for each of the rows specified by index.
+        """
+        if not isinstance(item, numbers.Integral):
+            raise NotImplementedError
+        return self.iloc[item]
+
+    @property
+    def loc(self):
+        """
+        Not currently supported.
+        """
+        return self.GeoSeriesLocIndexer(self)
+
+    @property
+    def iloc(self):
+        """
+        Return the i-th row of the GeoSeries.
+        """
+        return self.GeoSeriesILocIndexer(self)
+
+    def __len__(self):
+        """
+        Returns the number of unique geometries stored in this cuGeoSeries.
+        """
+        return int(len(self.types))
+
+    @property
+    def points(self):
+        """
+        The Points column is a simple numeric column. x and y coordinates
+        can be stored either interleaved or in separate columns. If a z
+        coordinate is present, it will be stored in a separate column.
+        """
+        return self._points
+
+    @property
+    def multipoints(self):
+        """
+        The MultiPoints column is similar to the Points column with the
+        addition of an offsets column. The offsets column stores the comparable
+        sizes and coordinates of each MultiPoint in the cuGeoSeries.
+        """
+        return self._multipoints
+
+    @property
+    def lines(self):
+        """
+        LineStrings contain the coordinates column, an offsets column, and a
+        multioffsets column. The multioffsets column stores the indices of the
+        offsets that indicate the beginning and end of each MultiLineString
+        segment.
+        """
+        return self._lines
+
+    @property
+    def polygons(self):
+        """
+        Polygons contain the coordinates column, a rings column specifying
+        the beginning and end of every polygon, a polygons column specifying
+        the beginning, or exterior, ring of each polygon and the end ring.
+        All rings after the first ring are interior rings.  Finally a
+        multipolys column stores the offsets of the polygons that should be
+        grouped into MultiPolygons.
+        """
+        return self._polygons
+
+    def to_pandas(self, index=None, nullable=False):
+        """
+        Treats to_pandas and to_geopandas as the same call, which improves
+        compatibility with pandas.
+        """
+        return self.to_geopandas(index=index, nullable=nullable)
+
+    def to_geopandas(self, index=None, nullable=False):
+        """
+        Returns a new GeoPandas GeoSeries object from the coordinates in
+        the cuspatial GeoSeries.
+        """
+        output = []
+        if nullable is True:
+            raise ValueError("cuGeoSeries doesn't support N/A yet")
```

Comment:
nitpick, but the repr for NA
```suggestion
            raise ValueError("cuGeoSeries doesn't support <NA> yet")
```